src/callable.cpp

```diff
@@ -53,6 +53,16 @@ void Callable::call(Interpreter *interpreter, std::vector<std::unique_ptr<Expr>>
             interpreter->environment = interpreter->environment->get_enclosing();
             return;
         }
+        catch (BreakException e)
```

Comment:
So this file is only for function calls, I think you want to handle these errors in the `visit_block` function.

Reply:
Made a change. Not sure if it's the correct change. We can review this tomorrow.